linode/manager
# build(dep): M3-6231: Update Typescript to latest (`4.9.5`)

## Description 📝

- Updates our repo to use the latest Typescript version
  - `4.1.5` ➡️ `4.9.5`
- Updates `reselect` selectors to use type inference per their recommendation to fix errors
- Uses generics in some places were ramda is used to circumvent some existing typescript errors that tsc somehow skipped in `4.1.5` 
  - I tried updating ramada but It causes more problems than it fixes

## E2E Run 📷

![Screenshot 2023-03-09 at 1 31 47 PM](https://user-images.githubusercontent.com/115251059/224122914-a7072145-86ee-407f-98d5-002f761ab23f.jpg)

## How to test 🧪

- Test Cloud Manager in general
- Verify that type checking passes in your local environment (`yarn workspace linode-manager run typecheck`)
- Verify that that Github Actions pass (specifically unit tests and typecheck)
